type conversion as function call argument

The following program `sample.go` triggers a panic:

```go
package main

type F func() (int, error)

func f1() (int, error) { return 3, nil }

func f2(a string, f F) {
    c, _ := f()
    println(a, c)
}
    
func main() {
    f2("hello", F(f1))
}
```

Expected result:
```console
$ go run ./sample.go
hello 3
```

Got:
```console
$ yaegi ./sample.go
./sample.go:13:2: panic
run: runtime error: index out of range [1] with length 1
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0002f4600, 0x40)
        /usr/lib/go/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc000217c70)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/interp.go:480 +0xc5
panic(0xea6880, 0xc00011e460)
        /usr/lib/go/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc0003780a0, 0xc000479b00, 0xc000217a60)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:171 +0x247
panic(0xea6880, 0xc00011e460)
        /usr/lib/go/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.call.func2(0xc0003780a0, 0xc0001141d8, 0x196, 0xea7760)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:907 +0x55
github.com/traefik/yaegi/interp.call.func7(0xc0003780a0, 0xc00011c500)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:1094 +0x925
github.com/traefik/yaegi/interp.runCfg(0xc000479b00, 0xc0003780a0)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:177 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc000372000, 0xc000479100, 0xc000378000)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:108 +0x2af
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc000372000, 0xc0002dc630, 0xb0, 0x7fff343565a2, 0x19, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/interp.go:572 +0x809
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc000372000, 0x7fff343565a2, 0x19, 0xc0002dc500, 0xb0, 0x0, 0x0, 0xc000121040)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/interp.go:394 +0xfa
main.runFile(0xc000372000, 0x7fff343565a2, 0x19, 0xc00010e240, 0x1)
        /home/marc/go/src/github.com/traefik/yaegi-master/cmd/yaegi/run.go:123 +0xae
main.run(0xc00000e090, 0x1, 0x1, 0x13, 0x13)
        /home/marc/go/src/github.com/traefik/yaegi-master/cmd/yaegi/run.go:89 +0x7e5
main.main()
        /home/marc/go/src/github.com/traefik/yaegi-master/cmd/yaegi/yaegi.go:144 +0x415
```

The problem seems to be that `F(f1)`, a type conversion expression is mis-matched as a function call expression in the input parameter processing.

--- a/_test/fun26.go
+++ b/_test/fun26.go
@@ -1,0 +1,17 @@
+package main
+
+type F func() (int, error)
+
+func f1() (int, error) { return 3, nil }
+
+func f2(a string, f F) {
+	c, _ := f()
+	println(a, c)
+}
+
+func main() {
+	f2("hello", F(f1))
+}
+
+// Output:
+// hello 3

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -2333,7 +2333,7 @@ func isCall(n *node) bool {
 }
 
 func isBinCall(n *node) bool {
-	return n.kind == callExpr && n.child[0].typ.cat == valueT && n.child[0].typ.rtype.Kind() == reflect.Func
+	return isCall(n) && n.child[0].typ.cat == valueT && n.child[0].typ.rtype.Kind() == reflect.Func
 }
 
 func mustReturnValue(n *node) bool {
@@ -2349,7 +2349,7 @@ func mustReturnValue(n *node) bool {
 }
 
 func isRegularCall(n *node) bool {
-	return n.kind == callExpr && n.child[0].typ.cat == funcT
+	return isCall(n) && n.child[0].typ.cat == funcT
 }
 
 func variadicPos(n *node) int {

